src/sql/workbench/contrib/notebook/test/browser/MarkdownTextTransformer.test.ts

```diff
@@ -0,0 +1,200 @@
+import { MockContextKeyService } from 'vs/platform/keybinding/test/common/mockKeybindingService';
+import { TestThemeService } from 'vs/platform/theme/test/common/testThemeService';
+import { TextModel } from 'vs/editor/common/model/textModel';
+import { isUndefinedOrNull } from 'vs/base/common/types';
+import { IAccessibilityService } from 'vs/platform/accessibility/common/accessibility';
+import { IContextKeyService } from 'vs/platform/contextkey/common/contextkey';
+import { TestCodeEditorService } from 'vs/editor/test/browser/editorTestServices';
+import { ICodeEditorService } from 'vs/editor/browser/services/codeEditorService';
+import { DefaultEndOfLine } from 'vs/editor/common/model';
+import { UndoRedoService } from 'vs/platform/undoRedo/common/undoRedoService';
+import { IThemeService } from 'vs/platform/theme/common/themeService';
+import { TestNotificationService } from 'vs/platform/notification/test/common/testNotificationService';
+import { TestDialogService } from 'vs/platform/dialogs/test/common/testDialogService';
+import { NotebookEditorStub, CellEditorProviderStub } from 'sql/workbench/contrib/notebook/test/stubs';
+import { QueryTextEditor } from 'sql/workbench/browser/modelComponents/queryTextEditor';
+import { NullTelemetryService } from 'vs/platform/telemetry/common/telemetryUtils';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { TestStorageService } from 'vs/workbench/test/common/workbenchTestServices';
+import { TestConfigurationService } from 'vs/platform/configuration/test/common/testConfigurationService';
+import { IEnvironmentService } from 'vs/platform/environment/common/environment';
+import { IStorageService } from 'vs/platform/storage/common/storage';
+
+class TestNotebookEditor extends NotebookEditorStub {
+	constructor(private _cellGuid?: string, private _instantiationService?: IInstantiationService) {
+		super();
+	}
+	cellEditors: CellEditorProviderStub[] = [new TestCellEditorProvider(this._cellGuid, this._instantiationService)];
+}
+
+class TestCellEditorProvider extends CellEditorProviderStub {
+	private _editor: QueryTextEditor;
+	private _cellGuid: string;
+	constructor(cellGuid: string, instantiationService?: IInstantiationService) {
+		super();
+		let div = dom.$('div', undefined, dom.$('span', { id: 'demospan' }));
+		let firstChild = div.firstChild as HTMLElement;
+
+		this._editor = new QueryTextEditor(
+			NullTelemetryService,
+			instantiationService,
+			new TestStorageService(),
+			new TestTextResourceConfigurationService(),
+			new TestThemeService(),
+			new TestEditorGroupsService(),
+			new TestEditorService(),
+			new TestConfigurationService()
+		);
+		this._editor.create(firstChild);
+		this._cellGuid = cellGuid;
+	}
+	cellGuid(): string {
+		return this._cellGuid;
+	}
+	getEditor(): QueryTextEditor {
+		return this._editor;
+	}
+}
+
+suite('MarkdownTextTransformer', () => {
+	let mockNotebookService: TypeMoq.Mock<INotebookService>;
```

Comment:
It'd be good to use the actual hook functions like before for this kind of setup so it's clear when it should be ran